lib/repack.c

```diff
@@ -794,7 +807,7 @@ repack_indexdef(PG_FUNCTION_ARGS)
 	/* specify the new tablespace or the original one if any */
 	if (tablespace || stmt.tablespace)
 		appendStringInfo(&str, " TABLESPACE %s",
-			(tablespace ? NameStr(*tablespace) : stmt.tablespace));
+			(tablespace ? quote_identifier(NameStr(*tablespace)) : stmt.tablespace));
```

Comment:
This looks like a bug of its own. Have we got an item like this in the unit test?

Reply:
I'll add tests to cover this case.